GitExtUtils/GitUI/TaskManager.cs

```diff
@@ -67,5 +61,36 @@ public void JoinPendingOperations()
             // since the latter adds new tasks to the collection and would therefore never complete.
             JoinableTaskContext.Factory.Run(_joinableTaskCollection.JoinTillEmptyAsync);
         }
+
+        public async Task ReportExceptionOnMainThreadAsync(Exception ex)
+        {
+            await JoinableTaskFactory.SwitchToMainThreadAsync();
+            Application.OnThreadException(ex.Demystify());
+        }
+
+        public void RunAsyncAndForget(Func<Task> asyncAction)
+        {
+            JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await TaskScheduler.Default;
```

Comment:
Why is this needed?
Is this why ConfigureAwait(false); is skipped, is the handling the same?
This seem to be the only functional change in this PR, what usage is needed?

Reply:
> Why is this needed?
> Is this why ConfigureAwait(false); is skipped,

yes

> is the handling the same?

I expect so.

> This seem to be the only functional change in this PR, what usage is needed?

The behavior should be the same - except that `TaskManager.RunAsyncAndForget(asyncAction)` does not call `JoinableTaskFactory.RunAsync()` twice any longer.

Reply:
IIRC `ConfigureAwait(false)` tells the scheduler to run continuation on any available thread. `await TaskScheduler.Default;` pushes the continuation to a background thread. So, the behaviour should be the same.